ucsmsdk_samples/firmware/hostfirmwarepack.py

```diff
@@ -261,11 +261,17 @@ def hfp_sp_attach(handle, sp_dn, hfp_name):
     if sp is None:
         raise ValueError("sp does not exist.")
 
-    org_dn = os.path.dirname(sp_dn)
-    dn = org_dn + "/fw-host-pack-" + hfp_name
-    obj = handle.query_dn(dn)
-    if obj is None:
-        raise ValueError("host firmware pack policy does not exist.")
+    obj = None
+    org_dn = os.path.dirname(sp.dn)
+    while obj is None:
+        dn = org_dn + "/fw-host-pack-" + hfp_name
+        obj = handle.query_dn(dn)
+        if obj:
+            break
+        elif obj is None and org_dn == 'org-root':
+            raise ValueError("host firmware pack policy does not exist.")
+        else:

```

Comment:
@ryaeng very minor.. but this else is not required, the statement under-else could be pulled one level up.